src/DistributedLock.Postgres/PostgresDistributedLock.Extensions.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Medallion.Threading.Internal;
+using System.Data;
+
+namespace Medallion.Threading.Postgres;
+
+public partial class PostgresDistributedLock
+{
+    public static bool TryAcquireWithTransaction(PostgresAdvisoryLockKey key, IDbTransaction transaction, TimeSpan timeout = default, CancellationToken cancellationToken = default)
+    {
+        if (key == null) { throw new ArgumentNullException(nameof(key)); }
+        if (transaction == null) { throw new ArgumentNullException(nameof(transaction)); }
+
+        var connection = new PostgresDatabaseConnection(transaction);
+
+        var handle = DistributedLockHelpers.TryAcquire(PostgresAdvisoryLock.ExclusiveLock, connection, key.ToString(), timeout, cancellationToken);
+
+        return handle != null;
+    }
+
+    public static void AcquireWithTransaction(PostgresAdvisoryLockKey key, IDbTransaction transaction, TimeSpan? timeout = null, CancellationToken cancellationToken = default)
+    {
+        if (key == null) { throw new ArgumentNullException(nameof(key)); }
+        if (transaction == null) { throw new ArgumentNullException(nameof(transaction)); }
+
+        var connection = new PostgresDatabaseConnection(transaction);
+
+        DistributedLockHelpers.Acquire(PostgresAdvisoryLock.ExclusiveLock, connection, key.ToString(), timeout, cancellationToken);
+    }
+
+    public static ValueTask<bool> TryAcquireWithTransactionAsync(PostgresAdvisoryLockKey key, IDbTransaction transaction, TimeSpan timeout = default, CancellationToken cancellationToken = default)
+    {
+        if (key == null) { throw new ArgumentNullException(nameof(key)); }
+        if (transaction == null) { throw new ArgumentNullException(nameof(transaction)); }
+
+        var connection = new PostgresDatabaseConnection(transaction);
```

Comment:
I feel like we can dispose this object after the acquire operation (just move the creation inside the helper function and add an async using block).

Reply:
I agree, it shouldn't impact anything regarding the external connection, so added disposal